### PR TITLE
[hab] add `hab pkg provides` command

### PR DESCRIPTION
**Pull request by metadave**
This PR implements the `hab pkg provides` subcommand, which gives us a _platform independent_ way to search **installed** Habitat packages for a given file.

Current flags are:

`-r` shows fully qualified package names
`-p` show full path to a found file

Here's some sample output searching for a binary, a header file, and a .so:

```
$ hab pkg provides bash
core/busybox-static
$ hab pkg provides -r bash
core/busybox-static/1.24.2/20160708162350
$ hab pkg provides -p bash
core/busybox-static: /hab/pkgs/core/busybox-static/1.24.2/20160708162350/bin/bash
$ hab pkg provides -rp bash
core/busybox-static/1.24.2/20160708162350: /hab/pkgs/core/busybox-static/1.24.2/20160708162350/bin/bash


$ hab pkg provides stdlib.h
core/glibc
$ hab pkg provides -r stdlib.h
core/glibc/2.22/20160612063629
$ hab pkg provides -p stdlib.h
core/glibc: /hab/pkgs/core/glibc/2.22/20160612063629/include/stdlib.h
core/glibc: /hab/pkgs/core/glibc/2.22/20160612063629/include/bits/stdlib.h
$ hab pkg provides -rp stdlib.h
core/glibc/2.22/20160612063629: /hab/pkgs/core/glibc/2.22/20160612063629/include/stdlib.h
core/glibc/2.22/20160612063629: /hab/pkgs/core/glibc/2.22/20160612063629/include/bits/stdlib.h

$ hab pkg provides -p libstdc++.so.6
core/gcc-libs: /hab/pkgs/core/gcc-libs/5.2.0/20160612075020/lib/libstdc++.so.6
```

cc @stevendanna

### CODE REVIEW

**Comment by thesentinels** [comment]
@metadave, thanks for your PR! By analyzing the annotation information on this pull request, we identified @fnichol, @reset, @seth, @juliandunn and @adamhjk to be potential reviewers

File: components/hab/src/command/pkg.rs [hunk 1]
```diff
@@ -290,6 +290,71 @@ pub mod path {
...
+                    }
+                }
+            }
+        }
+        // if we're not using full_path, then using a set will filter out
+        // duplicates. This shows the filtered set of matches
+        for entry in &found {
+            println!("{}", entry);
+        }
+        Ok(())
```
Comment (smith) [inline components/hab/src/command/pkg.rs]: Can we make this so it exits 1 if there are no matches? That's how `grep`, `dpkg -S`, `which`, and other tools behave.

Comment (metadave) [inline components/hab/src/command/pkg.rs]: fixed

**Comment by smith** [comment]
gif-keyboard-7700381563698830564

@thesentinels r+

**Comment by thesentinels** [comment]
:pushpin: Commit 59b5262 has been approved by `smith`

**Comment by thesentinels** [comment]
:hourglass: Testing commit 59b5262 with merge 5198fb4...

**Comment by thesentinels** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/habitat-sh/habitat/builds/147575389)

**Pull request merged**